K2Bridge/Factories/BucketFactory.cs

```diff
@@ -46,7 +52,78 @@ public static DateHistogramBucket CreateDateHistogramBucketFromDataRow(DataRow r
+                // Create new Regex to match the ID%PERCENTILE - example: 1%50.0
+                // var regex = new Regex(@"^(\d+)%{1}(100\.0|[0-9]?[0-9]\.[0-9]{1})$");
+                // var match = regex.Match(clmn.ColumnName);
+                var columnNameInfo = clmn.ColumnName.Split('%');
+
+                if (columnNameInfo.Length > 1)
+                {
+                    // key%metric%value1%value2%keyed
+                    var key = columnNameInfo[0];
+                    var metric = columnNameInfo[1];
+                    var queryValues = columnNameInfo[2..^1];
+                    var keyed = bool.Parse(columnNameInfo[^1]);
+
+                    if (metric == "percentile")
+                    {
+
+                        dhb.Aggs[key] = new Dictionary<string, object>();
+                        var returnValues = new Dictionary<string, double>();
+
+                        var percentileValues = (JArray)row[clmn.ColumnName];
+
+                        for (var index = 0; index < queryValues.Length; index++)
+                        {
+                            var name = queryValues[index];
+                            var value = percentileValues[index].ToString();
+                            returnValues.Add(name, double.Parse(value, CultureInfo.InvariantCulture));
+                        }
+
+                        if (keyed)
+                        {
+                            // keyed ====> Dictionary
+                            dhb.Aggs[key].Add("values", returnValues);
+                        }
+                        else
+                        {
+                            // not keyed ====> List(key,value)
+                            dhb.Aggs[key].Add("values", returnValues.ToList());
+                        }
+                    }
+
+                    /* ----> var rowss = row[clmn.ColumnName];
+                    // // var percentile = match.Groups[2].Value;
+                    // // logger.LogTrace("Defining the Percentile {percentile}", percentile);
+
+                    // // if (!dhb.Aggs.ContainsKey(match.Groups[1].Value))
+                    // // {
+                    // // dhb.Aggs[match.Groups[1].Value] = new Dictionary<string, double>();
+                    // //}
+
+                    // if (keyed) {
+                    //     // dictionnary
+                    //     dhb.Aggs[key] = new Dictionary<string, double>();
+
+                    //     //foreach
+                    //     foreach(var percentile in columnNameInfo[1..^1]) {
+                    //         dhb.Aggs[key].Add(percentile, )
+                    //     }
+                    // }
+                    // else {
+                    //     // array
+                    // }
+                    // dhb.Aggs[match.Groups[1].Value].Add(percentile, Convert.ToDouble(row[clmn.ColumnName]));*/
+                }
+                else
```

Comment:
nit: you can reduce the nesting by returning dhb here and remove the else

Reply:
(depends how nested it looks after you cleanup this file)

Reply:
return dhb is made after the foreach loop